src/AspNet.Security.OAuth.Vimeo/VimeoAuthenticationHandler.cs

```diff
@@ -46,17 +51,14 @@ public VimeoAuthenticationHandler([NotNull] HttpClient client)
 
             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
 
-            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, VimeoAuthenticationHelper.GetIdentifier(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim("urn:vimeo:fullname", VimeoAuthenticationHelper.GetFullName(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim("urn:vimeo:profileurl", VimeoAuthenticationHelper.GetProfileUrl(payload), Options.ClaimsIssuer);
-
             var principal = new ClaimsPrincipal(identity);
-            var ticket = new AuthenticationTicket(principal, properties, Options.AuthenticationScheme);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload);
+            context.RunClaimActions(payload);
 
-            var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens, payload);
-            await Options.Events.CreatingTicket(context);
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, payload.Value<string>("uri")?.Split('/')?.LastOrDefault(), ClaimValueTypes.String, Options.ClaimsIssuer));
```

Comment:
Same remark (I see this pattern in a few places, that would be cool to move to the claims actions API).